parmed/openmm/parameters.py

```diff
@@ -178,10 +179,16 @@ def _write_omm_atom_types(self, dest):
         dest.write(' </AtomTypes>\n')
 
     def _write_omm_residues(self, dest):
+        if not self.residues: return
         dest.write(' <Residues>\n')
         for name, residue in iteritems(self.residues):
             if not isinstance(residue, ResidueTemplate):
                 continue
+            # if AtomTypes are available, this will throw an exception if the
+            # residue is claiming a Type that doesn't exist
+            if self.atom_types:
```

Comment:
> Would there be any use case for having the Containers part of the OpenMMParameterSet in the future?

No -- Containers are really only for solvent boxes or multi-residue templates (which I've only ever seen Amber use for solvent boxes).  OpenMM would never make use of that.

> I don't see much difference between removing these at calling OpenMMParameterSet.from_parameterset or at writing residues to be fair - but maybe that would make the code a bit cleaner.

This assumes that the only thing you will ever *want* to do is write out an XML file.  If the OpenMMParameterSet does not really understand or deal with solvent boxes, then it should never inherit one from any kind of constructor IMO.  If you filter it out at creation time, you never have to protect against their inclusion anywhere else.

> have the self.typeify_templates call but add a if not self.atom_types: return line to the typeify_templates function (so if no Atom Types available - assume user takes the risk of some atom types possibly being wrong, if any Atom Types available - assume user wants the check). But if feeding the atom.atom_type 's is useless like I think - maybe we should just replace that with an assert test? @ChayaSt how does this work for you? She has the stream files, and it seems like they might contain some Atom Types but not all? In this case this wouldn't work and we should maybe just get rid of typeify_templates completely and accept the risk?

OK, `typeify_templates` is just causing problems for now.  This was really added with an eye to the future where ParmEd can do its own parameter assignment (like leap and ForceField).  And `atom_type` provides the same kind of type/class dichotomy that OpenMM and Tinker supports (and GROMACS, too, although it is used a little less often).  But as you point out, it doesn't have much use now outside of error-checking for the use of uninitialized/undefined types.  And it would be good to support *just* reading residue templates and not parameters.

Let's keep the method around, but the changes here that get rid of the explicit call to the method are probably the best way to go for now.

> and remove the ResidueTemplateContainer instances at calling OpenMMParameterSet.from_parameterset

:+1: 

> How does this sound?

Good!